docs/docs/docs/scala-3-derivation.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: docs
+title: Scala 3 Derivation
+---
+
+## {{page.title}}
+
+Scala 3 introduced native support for
+[type class derivation](https://dotty.epfl.ch/docs/reference/contextual/derivation.html) - a way to automatically
+generate type class instances for enums, enum cases, case classes, case objects and sealed traits that only have the
+aforementioned types as children. Instances are generated on demand by adding a `derives` clause on data types.
+
+PureConfig has some limited support for derivation of `ConfigReader` instances. In order to enable it, the relevant
+extensions must be imported:
+
+```scala
```

Comment:
Unfortunately we can't run any of this through mdoc as our website depends on `pureconfig-generic`, which isn't available in Scala 3. Instead, I ran this on `core/console` and copied the outputs here.